### ISSUE TITLE
weird dependency error

### ISSUE THREAD

**Issue by kumbafu** [NUARIG/surveyor#73]
in the following setup:

survey "dependency test" do
  section "section 1" do

```
q_copd_sh_1 "Have you ever smoked cigarettes?",:pick=>:one,:help_text=>"NO means less than 20 packs of cigarettes or 12 oz. of tobacco in a lifetime or less than 1 cigarette a day for 1 year."
  a_1 "Yes"
  a_2 "No"


q_copd_sh_1a "How old were you when you first started smoking cigarettes?", :help_text=>"age in years"
  a :integer
dependency :rule => "A"
condition_A :q_copd_sh_1, "==", :a_1

q_copd_sh_1b "Do you currently smoke cigarettes?",:pick=>:one, :help_text=>"as of 1 month ago"
  a_1 "Yes"
  a_2 "No"
dependency :rule => "B"
condition_B :q_copd_sh_1, "==", :a_1

a_copd_sh_1c "On the average of the entire time you smoked, how many cigarettes did you smoke per day?"
  a :integer
  dependency :rule => "A"
  condition_A :q_copd_sh_1, "==", :a_1

q_copd_sh_1bb "How many cigarettes do you smoke per day now?",:pick=>:one
  a_2 "integer"
dependency :rule => "C"
condition_C :q_copd_sh_1b, "==", :a_1


q_copd_sh_1ba "How old were you when you stopped?"
  a "Years",:integer
dependency :rule => "D"
condition_D :q_copd_sh_1b, "==", :a_2
```

  end
end

question   q_copd_sh_1ba  ends up with the following dependency:

   dependency :rule => "A"
    condition_A :q_copd_sh_1, "==", :a_1

**Comment by yoon** [comment]
"a_copd_sh_1c" should be "q_copd_sh_1c"

Also "q_copd_sh_1bb" should not have :pick => :one (although that doesn't affect the error)

**Comment by yoon** [comment]
test dependency issues. closed by e8b455a13a9d13bab54dbc2c45bb370d6967a48b
